### PR TITLE
Dockerize sc-machine

### PR DESCRIPTION
**Pull request by FallenChromium**
* [x] Read PR [documentation](https://github.com/ostis-ai/sc-machine/blob/main/docs/dev/pr.md/)
* [x] Update changelog
* [x] Update documentation

### CODE REVIEW

File: scripts/install_deps_ubuntu.sh [hunk 1]
```diff
@@ -1,12 +1,79 @@
...
+    libboost-system-dev
+    libboost-filesystem-dev
+    libboost-program-options-dev
+    libboost-python-dev
+    python3
+    python3-pip
+)
+
+packagelist_dev=(
+    qtbase5-dev
```
Comment (NikitaZotov) [inline scripts/install_deps_ubuntu.sh:30]:
```suggestion
```

File: scripts/install_deps_ubuntu.sh [hunk 2]
```diff
@@ -1,12 +1,79 @@
+#!/bin/bash
 APP_ROOT_PATH=$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && cd .. && pwd)
 
+function usage() {
+    cat <<USAGE
+
+    Usage: $0 [--rocksdb] [--dev]
+
+    Options:
+        --rocksdb:      installs deps required for RocksDB storage backend
```
Comment (NikitaZotov) [inline scripts/install_deps_ubuntu.sh:10]:
```suggestion
        --rocksdb:      installs deps required for rocksdb fs-storage backend
```

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.238.0/containers/cpp/.devcontainer/base.Dockerfile
+
+# [Choice] Debian / Ubuntu version (use Debian 11, Ubuntu 18.04/22.04 on local arm64/Apple Silicon): debian-11, debian-10, ubuntu-22.04, ubuntu-20.04, ubuntu-18.04
+ARG VARIANT="bullseye"
+FROM mcr.microsoft.com/vscode/devcontainers/cpp:0-${VARIANT}
```
Comment (NikitaZotov) [inline .devcontainer/Dockerfile:5]: It is needed only for vscode?

File: README.md [hunk 4]
```diff
@@ -1,92 +1,30 @@
...
+# Documentation: [GitHub Pages](https://ostis-ai.github.io/sc-machine)
 
-## Clone:
+# Quick start
+You can use our [OSTIS Web platform](https://github.com/ostis-ai/ostis-web-platform) project to get up and running quickly.
+# Installation
+## Docker
+We're providing a Docker image for this project. Head to [Installing with Docker](https://ostis-ai.github.io/sc-machine/docker) section of our docs to try it out!
+## Native
+Please refer to the docs for your OS: [Build on Linux](https://ostis-ai.github.io/sc-machine/build/linux-build/) and [Build on macOS](https://ostis-ai.github.io/sc-machine/build/osx-build/)
```
Comment (NikitaZotov) [inline README.md:11]:
```suggestion
Please, refer to the docs for your OS: [Build on Linux](https://ostis-ai.github.io/sc-machine/build/linux-build/) and [Build on macOS](https://ostis-ai.github.io/sc-machine/build/osx-build/)
```

Comment (NikitaZotov) [inline README.md:11]:
```suggestion
Please refer to the docs for your OS: [Build on Linux](https://ostis-ai.github.io/sc-machine/build/linux-build/) or [Build on macOS](https://ostis-ai.github.io/sc-machine/build/osx-build/)
```

File: README.md [hunk 5]
```diff
@@ -1,92 +1,30 @@
...
-### Usage example
-```sh
-cd sc-machine
-python3 scripts/build_kb.py ../ims.ostis.kb -c config/sc-machine.ini -o kb.bin
-```
-
-Default config file saves compiled KB one level above sc-machine, `sc-machine/..`
-
-Alternatively, you can use `build_kb.py` without a config file to define output path and other settings manually:
+For that, you can do the following:
```
Comment (NikitaZotov) [inline README.md:16]:
```suggestion
For that you can do the following:
```

File: README.md [hunk 6]
```diff
@@ -1,92 +1,30 @@
...
-!../ims.ostis.kb/ims/ostis_tech
-!../custom_kb/test.scs
-```
-
-## Servers
-
-Sc-machine provides network protocol to interact with **sc-server** - use `python3 scripts/run_sc_server.py -c config/sc-machine.ini` to run sc-server with the default config file.
-To get more information use `python3 scripts/run_sc_server.py -h`
+#build kb
+python3 scripts/build_kb.py -c config/sc-machine.ini <path to KB or repo.path file>
```
Comment (NikitaZotov) [inline README.md:19]: What is about make_all.sh script?

File: docs/build/linux-build.md [hunk 7]
```diff
@@ -1,4 +1,4 @@
-This instruction is intended to be used on Ubuntu.
+This instruction is intended to be used on Debian-based distros. For other distros, skip [Install dependencies](#install-dependencies) section and make sure you install all the required packages manually.
```
Comment (NikitaZotov) [inline docs/build/linux-build.md:1]:
```suggestion
This instruction is intended to be used on Debian-based distros. For other distros skip [Install dependencies](#install-dependencies) section and make sure you install all the required packages manually.
```

File: docs/other/repofile.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+## Knowledge base repository file - repo.path
+For more complex KB setups (with several folders across the filesystem, for example), we've created an internal file format called `repo.path`. It allows you to specify all the folders that should be built into the KB or exclude some files or subfolders
```
Comment (NikitaZotov) [inline docs/other/repofile.md:2]:
```suggestion
For more complex KB setups (with several folders across the filesystem, for example) we've created an internal file format called `repo.path`. It allows you to specify all the folders that should be built into the KB or exclude some files or subfolders
```

Comment (NikitaZotov) [inline docs/other/repofile.md:2]:
```suggestion
For more complex KB setups (with several folders across the filesystem, for example), we've created an internal file format called `repo.path`. It allows you to specify all the folders that should be built into the KB or exclude some files or subfolders.
```

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

**Review by NikitaZotov — APPROVED** [review]

**Review by VasKho — APPROVED** [review]

**Pull request merged**